src/server/api/url_parser.cpp

```diff
@@ -26,15 +26,15 @@ struct URLParser final : qi::grammar<Iterator, Into>
     {
         using boost::spirit::repository::qi::iter_pos;
 
-        alpha_numeral = qi::char_("a-zA-Z0-9");
+        identifier = qi::char_("a-zA-Z0-9_..--~:");
         percent_encoding =
             qi::char_('%') > qi::uint_parser<unsigned char, 16, 2, 2>()[qi::_val = qi::_1];
-        polyline_chars = qi::char_("a-zA-Z0-9_.--[]{}@?|\\~`^") | percent_encoding;
-        all_chars = polyline_chars | qi::char_("=,;:&().");
+        polyline_chars = qi::char_("a-zA-Z0-9_..--[]{}@?|\\~`^") | percent_encoding;
```

Comment:
Turns out we need to specify `..` for a single dot to work.

Reply:
I think a similar approach can be applied as above.

Perhaps out of scope, but I don't think `.` or `-` can appear in polylines, as their ASCII code is less than 63? Maybe that's why we've never noticed that the `.--` range doesn't work.